app/models/zone.rb

```diff
@@ -48,7 +55,38 @@ def find_master_server
     active_miq_servers.detect(&:is_master?)
   end
 
+  def self.create_maintenance_zone
```

Comment:
What do you think of something simpler like:
```ruby
def self.maintenance_zone
  @maintenance_zone ||= MiqRegion.my_region.maintenance_zone || create_maintenance_zone
end

def self.create_maintenance_zone
  name = "#{MAINTENANCE_ZONE_NAME_PREFIX}#{SecureRandom.uuid}"
  _log.info("Creating maintenance zone #{name}...")
  create!(:name => name, :description => name, :visible => false).tap do |zone|
    MiqRegion.my_region.update_attributes(:maintenance_zone => zone)
  end
rescue ActiveRecord::RecordInvalid
  retry
end
```

Reply:
when we move create_maintenance_zone from `seed` to getter, then we should do the same for default_zone - to be consistent, what do you think?

`SecureRandom ` is possible too, I choose sequence to make specs more understandable and for reason below..

I'm not convinced about `retry` for some common exception. It increases the risk of indefinite loop when unrelated code can raise it too. With combination of creating from getter it seems dangerous to me.

Reply:
If the worry is that unrelated errors could trigger an infinite loop we could always do what we used to do with the host names in refresh which was check if the name column is the reason for the error:
```
begin
  zone = build(:name => name, ...)
  zone.save!
rescue ActiveRecord::RecordInvalid
  raise if zone.errors[:name].blank?
  retry
end

Reply:
We could also add a threshold to the retry but if we have more then one UUID collision..buy a lottery ticket ;)

Reply:
> then we should do the same for default_zone - to be consistent, what do you think?

Maybe? But that's out of scope for this PR